File: app/src/main/java/com/library/testmvp/mvp/LoginModel.java
package com.library.testmvp.mvp;


import com.common.http.XRetrofit;
import com.common.http.base.BaseResponse;
import com.common.http.bean.ExampleBean;
import com.common.http.helper.RequestBodyHelper;

import java.util.HashMap;

import io.reactivex.Observable;

/**
 * @Author: https://github.com/manitozhang
 * @Email: 1271396448@qq.com
 * <p>
 * @Desc:
 */

public class LoginModel implements LoginContract.Model {

    @Override
    public Observable<BaseResponse<ExampleBean>> login(HashMap<String, Object> map) {
        return XRetrofit.getApi().example3(RequestBodyHelper.getRequestBody(map));
    }
}

File: app/src/main/java/com/library/testmvp/mvp/LoginModel.kt
package com.library.testmvp.mvp

import com.common.http.XRetrofit.api
import com.common.http.base.BaseResponse
import com.common.http.bean.ExampleBean
import com.common.http.helper.RequestBodyHelper.getRequestBody
import io.reactivex.Observable
import java.util.*

/**
 * @Author: 张鹏飞
 * @Email: 1271396448@qq.com
 *
 *
 * @Desc:
 */
class LoginModel : LoginContract.Model {
    override fun login(map: HashMap<String, ExampleBean>): Observable<BaseResponse<ExampleBean>> {
        return api!!.example3(getRequestBody(map))
    }
}